src/blindpsbt.cpp

```diff
@@ -386,7 +386,7 @@ BlindingStatus BlindPSBT(PartiallySignedTransaction& psbt, std::map<uint32_t, st
         }
 
         // Handle issuances
-        if (input.m_issuance_value) {
+        if (input.m_issuance_value != std::nullopt || input.m_issuance_value_commitment.IsCommitment() || input.m_issuance_inflation_keys_amount != std::nullopt || input.m_issuance_inflation_keys_commitment.IsCommitment()) {
```

Comment:
The previous code missed cases where there was a re-issuance token but not an issuance. This is allowed, and should be accounted for.

Reply:
For complex/long-winded logic checks like this, it would be nice to have methods on the input such as:

```
bool PSBTInput::HasIssuance() const
{
    if (m_issuance_value != std::nullopt || m_issuance_value_commitment.IsCommitment()) {
        return true; // Blinded or unblinded issuance
    }
    if (m_issuance_inflation_keys_amount != std::nullopt || m_issuance_inflation_keys_commitment.IsCommitment()) {
         return true; // Re-issuance token without an issuance
    }
    return false; // No issuance or re-issuance present
}
```

This would make the underlying logic much clearer IMO.